File: CSharp/Shop/Customers/Customer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Shop.Customers {
    using Items;
    public class Customer {
        private int funds = 0;
        private List<ItemOld> items = new List<ItemOld>();
        public Customer(int funds) {
            this.funds = funds;
        }
        public int Funds {
            get { return funds; }
        }
        public List<ItemOld> Items {
            get { return items; }
        }

        public void Buy(Shop shop, string item) {
            ItemOld purchasedItem = shop.BuyOld(item, ref funds);
            items.Add(purchasedItem);
        }
        public void Return(Shop shop, ItemOld item) {
            shop.ReturnOld(item);
            items.Remove(item);
            funds += item.Value;
        }
    }
}

File: Java/TddShop/src/Customer/Customer.java
package Customer;
import Shop.*;
import Item.*;

import java.util.ArrayList;
import java.util.List;

public class Customer {
    private List<Item> items;
    private Wallet wallet;
    public Customer(Wallet wallet){
        this.wallet = wallet;
        items = new ArrayList<Item>();
    }
    public Wallet Wallet(){
        return wallet;
    }
    public List<Item> Items(){
        return items;
    }
    public void Buy(Shop shop, String name) throws NotEnoughFundsException, ItemNotAvailableException, CannotReturnException{
        Item i = shop.Buy(name, wallet);
        items.add(i);
    }
    public void Return(Shop shop, Item i) throws CannotReturnException{
        shop.Return(i);
        items.remove(i);
        wallet.Funds += i.Value();
    }
}
